tests/unit/meta/test_snap_yaml.py

```diff
@@ -608,6 +608,59 @@ def test_get_provider_content_directories(new_dir, mocker):
     ]
 
 
+@pytest.mark.parametrize("grade", ["devel", "stable"])
+def test_grade(grade, simple_project, new_dir):
+    """Use the grade provided by the project."""
+    snap_yaml.write(
+        project=simple_project(grade=grade),
+        prime_dir=Path(new_dir),
+        arch="amd64",
+        arch_triplet="x86_64-linux-gnu",
+    )
+    yaml_file = Path("meta/snap.yaml")
+    assert yaml_file.is_file()
+
+    data = yaml.safe_load(yaml_file.read_text())
+
+    assert data["grade"] == grade
+
+
+def test_grade_default(emitter, simple_project, new_dir):
+    """Grade should default to `stable` when not provided by the project."""
+    snap_yaml.write(
+        project=simple_project(),
+        prime_dir=Path(new_dir),
+        arch="amd64",
+        arch_triplet="x86_64-linux-gnu",
+    )
+    yaml_file = Path("meta/snap.yaml")
+    assert yaml_file.is_file()
+
+    data = yaml.safe_load(yaml_file.read_text())
+
+    assert data["grade"] == "stable"
+
+    emitter.assert_debug("Grade not specified, using default value 'stable'.")
+
+
+def test_grade_build_base_devel(emitter, simple_project, new_dir):
```

Comment:
You should also test the behaviour when `build_base="devel"` and `grade="stable"`

Reply:
Uhh nevermind. I looked below and see this is undefined behaviour because it should never make it this far